docs/ai/server-generated-id.md

```diff
@@ -0,0 +1,30 @@
+When a resource has a server-generated ID, we need to store the generated ID into the kubernetes object,
+so that we can associate the GCP object with the kubernetes object on subsequent reconciliations.
+
+Legacy controllers would write this ID to spec.resourceID,
+which is the same field that users can specify if they want to adopt an existing resource.
+However, this makes the field ownership of the spec.resourceID field unclear: did the user specify it, or did KCC write it back?
+
+For greenfield controllers, we are introducing status.externalRef.
+This field will consistently give the GCP resource ID, as well as indicating whether a resource is under KCC control.
+
+The question is whether we should also write to spec.resourceID.
+To avoid a behavioural change, when moving from terraform to direct, we will write to both status.externalRef _and_ spec.resourceID.
+We want to avoid two problems:
+
+* It should be possible to create a GCP resource with the direct controller and then revert back to the terraform controller (without creating a new GCP resource)
+
+* Tooling or other controllers should continue to see the resourceID in the existing spec.resourceID field.
```

Comment:
While we are encouraging ALL KCC users to switch their usage to status.externalRef, we should not be tieing their migration of when to instantaneously switch their application for a specific field to a particular KCC update. Especially as that KCC update may be outside their control and may occur without their being warned.